<launch>

  <arg name="gui" default="true" />

  <node name="bounding_box_array_publisher"
        pkg="jsk_recognition_utils" type="bounding_box_array_publisher.py">
    <rosparam>
      rate: 10
      frame_id: /map
      boxes:
        - dimension: [0.09, 0.13, 0.15]
          orientation: [0.0, 0.0, 0.0, 1.0]
          position: [1.3860276937484741, -0.3705306053161621, 0.5474169254302979]
        - dimension: [0.15, 0.09, 0.13]
          orientation: [0, 0, 0, 1]
          position: [1.4170238971710205, -0.3371482789516449, 0.5605951547622681]
    </rosparam>
  </node>
  <group if="$(arg gui)">
    <node name="rviz"
          pkg="rviz" type="rviz"
          args="-d $(find jsk_pcl_ros_utils)/sample/config/sample_evaluate_box_segmentation_by_gt_box.rviz">
    </node>
  </group>

  <node name="boxes_to_gt_box"
        pkg="jsk_pcl_ros_utils" type="bounding_box_array_to_bounding_box">
    <remap from="~input" to="bounding_box_array_publisher/output" />
    <rosparam>
      index: 0
    </rosparam>
  </node>
  <node name="boxes_to_box"
        pkg="jsk_pcl_ros_utils" type="bounding_box_array_to_bounding_box">
    <remap from="~input" to="bounding_box_array_publisher/output" />
    <rosparam>
      index: 1
    </rosparam>
  </node>

  <node name="evaluate_box_segmentation_by_gt_box"
        pkg="jsk_pcl_ros_utils" type="evaluate_box_segmentation_by_gt_box.py"
        respawn="true">
    <remap from="~input/box_gt" to="boxes_to_gt_box/output" />
    <remap from="~input/box" to="boxes_to_box/output" />
  </node>

</launch>
